crates/viewer/re_space_view_spatial/src/proc_mesh.rs

```diff
@@ -128,13 +211,173 @@ fn generate_wireframe(key: &ProcMeshKey, render_ctx: &RenderContext) -> Wirefram
                     .collect()
             };
             WireframeMesh {
-                // sphere’s radius is 1, so its size is 2
-                bbox: re_math::BoundingBox::from_center_size(Vec3::splat(0.0), Vec3::splat(2.0)),
+                bbox: key.simple_bounding_box(),
                 vertex_count: line_strips.iter().map(|v| v.len()).sum(),
                 line_strips,
             }
         }
     }
 }
 
-// TODO(kpreid): A solid (rather than wireframe) mesh cache implementation should live here.
+// ----------------------------------------------------------------------------
+
+/// Cache for the computation of triangle meshes from [`ProcMeshKey`]s that depict the
+/// shape as a solid object.
+#[derive(Default)]
+pub struct SolidCache(ahash::HashMap<ProcMeshKey, Option<SolidMesh>>);
+
+impl SolidCache {
+    pub fn entry(&mut self, key: ProcMeshKey, render_ctx: &RenderContext) -> Option<SolidMesh> {
+        re_tracing::profile_function!();
+
+        self.0
+            .entry(key)
+            .or_insert_with(|| {
+                re_log::debug!("Generating mesh {key:?}…");
+
+                match generate_solid(&key, render_ctx) {
+                    Ok(mesh) => Some(mesh),
+                    Err(err) => {
+                        re_log::warn!(
+                            "Failed to generate mesh {key:?}: {}",
+                            re_error::format_ref(&err)
+                        );
+                        None
+                    }
+                }
+            })
+            .clone()
+    }
+}
+
+impl Cache for SolidCache {
+    fn begin_frame(&mut self) {}
+
+    fn purge_memory(&mut self) {
+        self.0.clear();
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+}
+
+/// Generate a solid triangle mesh without caching.
+fn generate_solid(
```

Comment:
We recently vendored `macaw` as `re_math` which already comes with mesh generation methods https://github.com/rerun-io/re_math/blob/main/src/mesh_gen.rs I think we should prefer those over the hand-rolled here.

As pointed out [here](https://github.com/rerun-io/rerun/pull/6853#discussion_r1675413559), we'll stick with hexasphere for the moment, but that should eventually also go through the `re_math` mesh generation, so we don't have to duplicate glam etc. (also right now the wireframe is obviously using hexasphere as well and we have to align these)

Reply:
I've done this and it works nicely (changes not pushed yet, because I'm still working on sorting out the big stuff). Of course the wireframe mesh still needs custom code because `MeshGen` doesn't cover that. Thanks for making this connection.

Reply:
Done and pushed.